README.md

```diff
@@ -1,40 +1,69 @@
+. The service uses the [`biocommons/hgvs`](https://github.com/biocommons/hgvs) library to convert
 HGVS to VCF. To start the translator service:
+
 ```
 > cd docker
 > docker-compose up
 ```
+
 To see it in action, post an array of HGVS strings to the h2v endpoint:
+
 ```
 curl -H 'Content-Type: application/json' -d '["NM_000088.3:c.589G>T", "NC_000017.10:g.48275363C>A"]' 'http://localhost:1234/h2v?keep_left_anchor=False'
 ```
-These string can either be genomic HGVS (`NC_000017.10:g.48275363C>A`) or transcript HGVS (`NM_000088.3:c.589G>T`). 
-Two flags can be set for this service: 
+
+These string can either be genomic HGVS (`NC_000017.10:g.48275363C>A`) or transcript
+HGVS (`NM_000088.3:c.589G>T`). Two flags can be set for this service:
 
 | Flag | Usage | Example |
 | ---- | ----- | ------- |
-| `keep_left_anchor` | The service can either return variants with the last unchanged nucleotide (=left anchor), or without it. Please note that this will go for all variants, including insertions for which an anchor is preferred. | `True`: `{"ref": "CT", "alt": "CC", "chrom":"19", "pos":"49473113", "type":"sub"}`   
+| `keep_left_anchor` | The service can either return variants with the last unchanged nucleotide (=left anchor), or without it. Please note that this will go for all variants, including insertions for which an anchor is preferred. | `True`: `{"ref": "CT", "alt": "CC", "chrom":"19", "pos":"49473113", "type":"sub"}`
 |      |        | `False`: `{"ref": "T", "alt": "C", "chrom":"19", "pos":"49473114", "type":"sub"}` |
 | `strict` | The strict mode of the HGVS libary will be applied when strict is set to `True`. This means that validation warnings will result in errors rather than just warnings. ||
 
 ## file processing pipeline
+
 [Spring Boot](https://spring.io/projects/spring-boot) application that uses
 [Apache Camel](http://camel.apache.org/) to read input files from `src/test/inbox`
 and annotate the lines with vcf info retrieved from the hgvs annotator.
 
-Results will be stored in results dir once finished.
-Results with an error message will be routed to error file.
+Results will be stored in results dir once finished. Results with an error message will be routed to
+error file.
+
+Requires JDK 8. To run the pipeline:
 
-Requires JDK 8.
-To run the pipeline:
 ```
 > mvn clean spring-boot:run
 ```
+
 ![Pipeline overview](./vkgl.svg)
+
+## Gene validation
+
+Validation and correction of provided HGNC symbols is done using an export downloaded from 
+[the biomart website](https://biomart.genenames.org/martform/#!/default/HGNC?datasets=hgnc_gene_mart). 
+The selected attributes are:
+- HGNC ID
+- Status
+- Approved symbol
+- Approved name
+- Alias symbol
+- Previous symbol
+- Chromosome
+- Chromosome location
+- Locus group
+- NCBI gene ID 
+- Ensembl gene ID
+- UCSC gene ID  
+
+The downloaded file is named `hgnc_genes.tsv` and is stored in `src/main/resources`. To have the
```

Comment:
Disadvantage of this approach is that you have to create a new release whenever HGNC is updated in order to use the latest HGNC information. How about providing the resource as an input pipeline parameter (default if no input file is provided: pipeline will attempt to download the latest version)?

Reply:
> Disadvantage of this approach is that you have to create a new release whenever HGNC is updated in order to use the latest HGNC information. How about providing the resource as an input pipeline parameter (default if no input file is provided: pipeline will attempt to download the latest version)?

I have thought about this, but we are not working with script parameters at the moment and I don't know how to implement it in combination with apache camel. Additionally I don't think it's possible to download this file programmatically.